### PR TITLE
[tune] Fixed wait_for_gpu to handle str representations of ordinal IDs

### PR DESCRIPTION
**Pull request by tgaddair**
## Why are these changes needed?

`wait_for_gpus` does not handle the case when the user passes in an ordinal GPU ID as a string, which is the format returned by `ray.get_gpu_ids()`.  As such, the default behavior of this function when `gpu_id=None` is currently broken.

## Related issue number

#13486.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [X] This PR is not tested :(

cc @richardliaw

### CODE REVIEW

File: python/ray/tune/utils/util.py [hunk 1]
```diff
@@ -503,34 +503,38 @@ def tune_func(config):
...
+    gpu_attr = 'id'
+    if isinstance(gpu_id, str):
+        try:
+            # GPU ID returned from `ray.get_gpu_ids()` is a str representation
+            # of the int GPU ID, so check for this case
+            gpu_id = int(gpu_id)
+        except ValueError:
+            # Could not coerce gpu_id to int, so assume UUID and compare against `uuid` attribute
+            # e.g., 'GPU-04546190-b68d-65ac-101b-035f8faed77d'
+            gpu_attr = 'uuid'
```
Comment (richardliaw) [inline python/ray/tune/utils/util.py:523]: how about:

```suggestion
    if isinstance(gpu_id, str):
        if gpu_id.isdigit():
            gpu_id = int(gpu_id)
        else:
            gpu_attr = "attr"
```

File: python/ray/tune/utils/util.py [hunk 2]
```diff
@@ -503,34 +503,38 @@ def tune_func(config):
...
+            "`wait_for_gpu` takes either GPU ordinal ID (e.g., '0') or "
+            "UUID (e.g., 'GPU-04546190-b68d-65ac-101b-035f8faed77d').")
 
     for i in range(int(retry)):
-        if isinstance(gpu_id, int):
-            gpu_object = [g for g in GPUtil.getGPUs() if g.id == gpu_id][0]
-        else:
-            gpu_object = [g for g in GPUtil.getGPUs() if g.uuid == gpu_id][0]
-
+        gpu_object = next(g for g in GPUtil.getGPUs() if gpu_id_fn(g) == gpu_id)
```
Comment (richardliaw) [inline python/ray/tune/utils/util.py:537]: nice...

File: python/ray/tune/utils/util.py [hunk 3]
```diff
@@ -503,34 +503,38 @@ def tune_func(config):
...
+        except ValueError:
+            # Could not coerce gpu_id to int, so assume UUID and compare against `uuid` attribute
+            # e.g., 'GPU-04546190-b68d-65ac-101b-035f8faed77d'
+            gpu_attr = 'uuid'
+
+    def gpu_id_fn(g):
+        # Returns either `g.id` or `g.uuid` depending on the format of the input `gpu_id`
+        return getattr(g, gpu_attr)
+
+    gpu_ids = {gpu_id_fn(g) for g in GPUtil.getGPUS()}
```
Comment (ANarayan) [inline python/ray/tune/utils/util.py:529]: I think there is a typo here: GPUtil.getGPUS() should be GPUtil.getGPUs()

**Review by richardliaw — APPROVED** [review]
Nice!

**Pull request merged**